Broadcast-Ext/OpenTok Live/SampleHandler.m

```diff
@@ -0,0 +1,215 @@
+//
+//  SampleHandler.m
+//  OpenTok Live
+//
+//  Created by Sridhar Bollam on 8/4/19.
+//  Copyright © 2019 TokBox, Inc. All rights reserved.
+//
+
+
+#import "SampleHandler.h"
+#import "OTBroadcastExtHelper.h"
+
+@implementation SampleHandler
+{
+    CVPixelBufferPoolRef _pixelBufferPool;
+    CVPixelBufferRef _pixelBuffer;
+    int64_t _num_frames;
+    CIContext * _ciContext;
+    CIFilter* _scaleFilter;
+    bool _capturing;
+    
+    OTBroadcastExtHelper *_broadcastHelper;
+    
+    dispatch_queue_t _capture_queue;
+    
+}
+
+- (void)broadcastStartedWithSetupInfo:(NSDictionary<NSString *,NSObject *> *)setupInfo {
+    
+    // User has requested to start the broadcast. Setup info from the UI extension can be supplied but optional.
+    
+    // Provide session id and token
+    _broadcastHelper = [[OTBroadcastExtHelper alloc] initWithPartnerId:@""
+                                                             sessionId: @""
+                                                              andToken:@""
+                                                         videoCapturer:self];
+    _capture_queue = dispatch_queue_create("com.tokbox.OTBroadcastVideoCapture",
+                                           DISPATCH_QUEUE_SERIAL);
+    
+    _num_frames = 0;
+    [self destroyPixelBuffers];
+    [_broadcastHelper connect];
+}
+
+- (void)broadcastPaused {
+    // User has requested to pause the broadcast. Samples will stop being delivered.
+}
+
+- (void)broadcastResumed {
+    // User has requested to resume the broadcast. Samples delivery will resume.
+}
+
+- (void)broadcastFinished {
+    // User has requested to finish the broadcast.
+    [_broadcastHelper disconnect];
+}
+
+- (void) processPixelBuffer:(CVPixelBufferRef)pixelBuffer timeStamp:(CMTime)ts
+{
+    CIImage *ciImage = [CIImage imageWithCVPixelBuffer:pixelBuffer
+                                               options:nil];
+    //CGFloat imageWidth = CVPixelBufferGetWidth(pixelBuffer);
+    //CGFloat imageHeight = CVPixelBufferGetHeight(pixelBuffer);
+    CGFloat aspectRatio = 1;//imageHeight / imageWidth;
+    
+    ciImage = [self scaleFilterImage:ciImage
+                     withAspectRatio:aspectRatio scale:0.60];
```

Comment:
why scaling it to 0.60?, Looks like a magic number

Reply:
The scaling 0.60 is tested on iPhone 8 and iPhone Xs to maintain < 50MB memory usage.

Reply:
I think we can specify it somehow, if we just put the number it can be confusing for the reader.